### ISSUE TITLE
Can we drag and drop event?

### ISSUE THREAD

**Issue by vinhtran6921** [hoangnm/react-native-week-view#14]
Can we implement drag and drop events in this calendar? Look like fullcalendar.io
I really appreciate your help.

**Comment by hoangnm** [comment]
Hi @vinhtran6921, I'll add the feature soon. Thanks for trying the lib.

**Comment by ymcewen** [comment]
@hoangnm did you get around to this? How about tapping on the grid to add a new event?

**Comment by pdpino** [comment]
Tapping on the grid is supported with the `onGridClick()` method

**Comment by rdewolff** [comment]
When do u plan to add this feature @hoangnm ?

**Comment by vipin-ios** [comment]
Hi 
is any update on this feature? when do you plan to add this please let us know.

**Comment by pdpino** [comment]
I can try to implement this feature, though in about 2-4 weeks :slightly_smiling_face:

**Comment by rdewolff** [comment]
That would be great!

**Comment by pdpino** [comment]
Update: I'm currently working on this, hope to have a first working version in a week or so

**Comment by rdewolff** [comment]
Amazing, looking forward seeing the results 😀

**Comment by pdpino** [comment]
I have a first working version :smiley: 

![rn-week-view-feature-drag-and-drop-2](https://user-images.githubusercontent.com/20805451/112730958-3cac2f80-8f13-11eb-8ddd-e24b242a227e.gif)

I updated the docs with the instructions on the two new props: `isDraggable` and `onDragEvent`.
Any comments/suggestions for functionality or documentation are welcomed!!

Some caveats:

* When you drag an event from left to right the vertical grid line is shown above the event; though from right to left the event is above the line (see the gif). The latter is the expected behavior, and the former is a bug (comes from the way we display the vertical lines in the first place). I tried to fix it using `zIndex` in the styles, but had no luck. I will try to fix this in a future PR.

* Regarding performance, the animation when the event is dropped is very smooth if there are few events. If there are many events saved (say, 2000), the animation may not be so smooth:

![rn-week-view-feature-drag-and-drop-lagged](https://user-images.githubusercontent.com/20805451/112731279-07a0dc80-8f15-11eb-816e-1de6c54bfb3d.gif)

This occurs because when the event is dropped, the `events` prop must be updated, which triggers a re-render on **all** the events in the array. It would be optimal if only the event changed gets re-rendered, but improving this is outside the scope of this PR. I will try to address this issue in the future.

**Comment by pbpranavk** [comment]
Hi @pdpino great work on the library, any update on this? when are you planning to release this?

**Comment by pdpino** [comment]
Yes, sorry! I've been very busy and this feature got a bit delayed, even though the code is _almost_ ready.

I will try to speed things up to release it :slightly_smiling_face:

**Comment by pbpranavk** [comment]
Hey @pdpino thanks again for your efforts, you are doing great stuff! 🙂
